docs/features/software-templates/migrating-from-v1alpha-to-v1beta2.md

```diff
@@ -0,0 +1,328 @@
+a much more extensible way. We now need to tell the scaffolder what to do with
+these parameters and what to do with the user input.
+
+We've made templating using `cookiecutter` a little more simpler. You don't need
+to store the `cookiecutter` skeleton in the same directory as the
+`template.yaml` definition, it can live wherever you like, maybe a shared
+repository somewhere so you can re-use the skeletons but apply different actions
+for different templates depending on your use case.
+
+We also no longer need to have a directory called
+`{{cookiecutter.component_id}}`. This is because now we can't ensure that
+`component_id` will be a parameter that is provided from the frontend, this
+could break `cookiecutter`. If your directory structure used to look like this:
+
+```
+my-awesome-template
+  -> {{cookiecutter.component_id}}
+    -> file.txt
+    -> some_more_files.ts
+  -> hooks
+    -> post_gen_project.sh
+  -> template.yaml
+```
+
+We now recommend that you move to the following structure:
+
+```
+my-awesome-template
+  -> skeleton
+    -> file.txt
+    -> some_more_files.ts
+  -> template.yaml
+```
+
+This migration renames the skeleton folder to something more semantic, and also
+drops support for `cookiecutter` hooks. We've dropped support for `cookiecutter`
+hooks for now, as hopefully everything that is stored in these hooks can be
+moved to `actions` instead, and for security reasons, it's a little more secure
+to run trusted code that you ship with Backstage as an action rather than some
+script that can be pulled in from anywhere which doesn't get vetted first. It's
+a pretty big security risk that those scripts will be run on Backstage instances
+inside your infrastructure, especially `.sh` files.
+
+If you really need hooks and can't find a suitable solution by using actions
+please reach out to use through a ticket and we can see what we can do :)
+
+You'll notice that we removed the `templater` property from the `spec`
+definition in the template `yaml`, so there's no way to define that this is a
+`cookiecutter` `templater`.
+
+We've created a built-in action that you can use which will when run, go grab a
+directory from anywhere and run `cookiecutter` on top of it, and then extract
+the contents into the working directory for the scaffolder.
+
+Adding the `steps` for a simple template should look something like the
+following:
+
+```yaml
+spec:
+  steps:
+    # this action will go use cookiecutter to template some files into the working directory
+    - id: template # an ID for the templating step
+      name: Create skeleton # A user friendly name for the action
+      action: fetch:cookiecutter
+      input:
+        url: ./skeleton # this is the directory for you skeleton files. If its located next to the `template.yaml` then you can use relative path, otherwise you can use absolute URL's that point at the VCS: https://github.com/backstage/backstage/tree/master/some_folder_somewhere
```

Comment:
```suggestion
        url: ./skeleton # this is the directory for your skeleton files. If it's located next to the `template.yaml` then you can use a relative path, otherwise you can use absolute URLs that point at the VCS: https://github.com/backstage/backstage/tree/master/some_folder_somewhere
```

Reply:
these long comments should probably be on their own line and broken up into several lines for rendering sanity